runtime/runtime/src/ext.rs

```diff
@@ -354,4 +359,27 @@ impl<'a> External for RuntimeExt<'a> {
     fn get_receipt_receiver(&self, receipt_index: ReceiptIndex) -> &AccountId {
         self.receipt_manager.get_receipt_receiver(receipt_index)
     }
+
+    fn code_hash(&self) -> CryptoHash {
+        self.account.code_hash()
+    }
+
+    fn get_contract(&self) -> Option<Arc<ContractCode>> {
+        let account_id = self.account_id();
+        let code_hash = self.code_hash();
+        let version = self.current_protocol_version;
+        if checked_feature!("stable", EthImplicitAccounts, self.current_protocol_version)
+            && account_id.get_account_type() == AccountType::EthImplicitAccount
+        {
+            let chain_id = self.chain_id();
+            assert!(&code_hash == wallet_contract_magic_bytes(&chain_id).hash());
+            return Some(wallet_contract(&chain_id));
+        }
+        let mode = match checked_feature!("stable", ChunkNodesCache, version) {
+            true => Some(TrieCacheMode::CachingShard),
+            false => None,
+        };
+        let _guard = self.trie_update.with_trie_cache_mode(mode);
```

Comment:
For here as well as in the other guard that can actually use `None` as an input here: What’s the point of having `with_trie_cache_mode` actually resetting the cache mode without changing it in the first place?

I’m thinking the two of them could be written like:
```rust
let _guard = match cache_enabled {
    true => Some(trie_update.with_trie_cache_mode(CachingShard)),
    false => None,
}
```

Which’d simplify the API for `with_trie_cache_mode` by removing an unused feature of it.

WDYT?

Reply:
So the use-case I was thinking of is that the code within the guard could arbitrarily `set_trie_cache_mode` and whatnot, and by the end of the guarded region we would still reset back to the original state. Kinda a guard-in-depth measure. Turned out to be unnecessary as `set_trie_cache_mode` was eventually deleted, but I figured that it would be alright to keep this measure in place since it was already implemented.

It is still somewhat possible to emulate `set_trie_cache_mode` with `mem::forget`ting the guard, although that would leak the reference count on the "enable switch" thingy. Which in turn might not be a huge deal since we don't anticipate to have very many of these switch during a lifetime of a process anyway...

Reply:
If this turns out to be a problem, changing this should be quite straightforward, so I'm going to leave as-is for the time being.